src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -261,7 +275,8 @@ public void unsubscribe() {
           }
         }
       }
-      processedRecords = _consumerRecordsProcessor.process(rawRecords);
+      _lastProcessedResult = _consumerRecordsProcessor.process(rawRecords);
```

Comment:
_lastProcessedResult will still have all the processed records in it which will be retained in-memory until the next call to pool.  Since this code also deals with large messages this may be quite a bit of retained memory.